packages/components/src/DataList/DataList.types.ts

```diff
@@ -97,6 +102,19 @@ export interface DataListProps<T extends DataListObject> {
    * The callback function when the user scrolls to the bottom of the list.
    */
   readonly onLoadMore?: () => void;
+
+  readonly onSortingChange?: (sorting: DataListSorting) => void;
+
+  /**
+   * `sortable`: List of keys that are sortable.
+   * `initialState`: The initial state of the sorting.
+   * `onSort`: The callback function when the user sorting a column.
+   */
+  readonly sorting?: {
+    sortable: string[];
```

Comment:
same here, i think you can narrow this down to whatever keys is available on the data/header?

Reply:
I narrowed it down, it gets only the headers' keys, but the `value` of `DataListContext.Provider` is complaining about it